container/utils.go

```diff
@@ -6,20 +6,39 @@ package container
 import (
 	"os/exec"
 	"runtime"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/core/instance"
 )
 
-var RunningInContainer = func() bool {
-	if runtime.GOOS != "linux" {
-		return false
+func pathForExcutable(executableName string) (string, error) {
+	var ePath string
+	if path, err := exec.LookPath(executableName); err == nil {
+		ePath = path
+	} else if path, err := exec.LookPath("/usr/bin/" + executableName); err == nil {
+		ePath = path
+	} else {
+		return "", errors.NotFoundf("%s executable", executableName)
 	}
-
-	/* running-in-container is in init-scripts-helpers, and is smart enough
-	 * to ask systemd whether it knows if the task is running in a container.
-	 */
-	cmd := exec.Command("running-in-container")
-	return cmd.Run() == nil
+	return ePath, nil
 }
 
-func ContainersSupported() bool {
-	return !RunningInContainer()
+func ContainersSupported() []instance.ContainerType {
```

Comment:
This should also just return an error, it makes calls that can fail for unexpected reasons, so it should actually error.